src/single/index.js

```diff
@@ -3,6 +3,32 @@
     var loadingTimer = 0;
 
     var infrastructure = {
+        buildElementsTree: function(entity) {
+            if (!entity) return null;
+
+            var i, j;
+            if (Array.isArray(entity)) {
+              // 我们约定，数组的第一个元素存储根节点的信息
+              var root = infrastructure.buildElementsTree(entity[0]);
+              for (i=1; i<entity.length; i++) {
+                  root.appendChild(infrastructure.buildElementsTree(entity[i]));
+              }
+              return root;
+            } else {
+                var element = document.createElement(entity.targetName);
```

Comment:
I think we could  replece "targetName" with "NodeName" in here, thus we can express means clearly.